packages/notebook/src/widget.ts

```diff
@@ -989,6 +991,7 @@ export class StaticNotebook extends WindowedList {
   private _idleCallBack: number | null;
   private _mimetype: string;
   private _mimetypeService: IEditorMimeTypeService;
+  readonly _history: INotebookHistory | undefined;
```

Comment:
I would recommend making this one protected

```suggestion
  protected readonly _history: INotebookHistory | undefined;
```

Reply:
We need a way to access history:

```
      ../notebook/src/actions.tsx(2100,20): error TS2445: Property '_history' is protected and only accessible within class 'StaticNotebook' and its subclasses.
```

Maybe it needs to be just `readolny` (public) but then it should probably not have `_` prefix. This is similar to the story with `footer`. Thoughts?

Reply:
Right, it used to be private and notebook had its own back/forward methods that called history from the `Notebook` class which were in turn called from `accessPreviousHistory` and `accessNextHistory` in [actions](https://github.com/jupyterlab/jupyterlab/blob/a37354a42b6d63f64dcb0e2a16216897b5919348/packages/notebook/src/actions.tsx#L2092). Now those just directly use `notebook._history`. So it seems we should either revert the code to keep `_history` private or just make it readonly and lose the `_` prefix. I would lean towards the second option.

Reply:
> I would lean towards the second option.

Let's do that to unblock this pull request. We will still have some time to change things before 4.1 release if we decide it was not the right call.